### PR TITLE
Add support for compatibility tools extensions

### PR DESCRIPTION
**Pull request by gasinvein**
This is changeset from #349 with merge conflicts resolved and "Proton" renamed to "CompatibilityTool".

### CODE REVIEW

File: steam_wrapper/steam_wrapper.py [hunk 1]
```diff
@@ -255,4 +292,5 @@ def main(steam_binary=STEAM_PATH):
     timezone_workaround()
     configure_shared_library_guard()
     enable_discord_rpc()
-    os.execve(steam_binary, [steam_binary] + sys.argv[1:], os.environ)
+    p = subprocess.Popen([steam_binary] + sys.argv[1:])
```
Comment (gasinvein) [inline steam_wrapper/steam_wrapper.py]: @nanonyme @refi64 If we can avoid IO-intensive copying of the whole extension, maybe we should switch back to using `os.execve`?

**Review by nanonyme — COMMENTED** [review]
If we can, we should use simple execv.

File: steam_wrapper/steam_wrapper.py [hunk 2]
```diff
@@ -243,10 +245,45 @@ def configure_shared_library_guard():
         if mode > 1:
             os.environ["LD_BIND_NOW"] = "1"
 
+def setup_compat_tool_extensions(current_info):
+    compat_tool_dest = Path('.local/share/Steam/compatibilitytools.d')
```
Comment (gasinvein) [inline steam_wrapper/steam_wrapper.py]: I wonder, why this path is relative?

**Comment by nanonyme** [comment]
Would you mind re-targeting against beta branch?

**Comment by gasinvein** [comment]
Should I open another PR, or just change base branch for this one?

**Comment by nanonyme** [comment]
I was intending to add all commits from beta to master by rebasing once 19.08 goes live so not really preferring to merge things into master anymore since will make life harder in one month.

**Comment by nanonyme** [comment]
I was intending to add all commits from beta to master by rebasing once 19.08 goes live so not really preferring to merge things into master anymore since will make life harder in one month.

**Comment by nanonyme** [comment]
Rebased on top of beta, will merge once build passes.

File: steam_wrapper/steam_wrapper.py (hunk 2 above)
Comment (nanonyme) [reply steam_wrapper/steam_wrapper.py]: Doesn't matter, the script initially chdirs to home to have consistent paths.

File: com.valvesoftware.Steam.yml [hunk 3]
```diff
@@ -75,6 +75,12 @@ add-extensions:
     download-if: active-gl-driver
     enable-if: active-gl-driver
 
+  com.valvesoftware.Steam.CompatibilityTool:
+    subdirectories: true
+    directory: compatibilitytools.d
```
Comment (gasinvein) [inline com.valvesoftware.Steam.yml]: Multiple 3rd-partiy compatibility tools install themselves to `$prefix/share/steam/compatibilitytools.d`. Maybe we should do the same?

Comment (nanonyme) [reply com.valvesoftware.Steam.yml]: I haven't actually tried these so I'm going to trust others on this for now. I guess that means when you create an extension, you would drop files into the root directory.

Comment (gasinvein) [reply com.valvesoftware.Steam.yml]: Well, I guess the mountpoint could be changed later anyway.

**Pull request merged**